pkg/controllers/deprovisioning/events/suite_test.go

```diff
@@ -0,0 +1,109 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package events_test
+
+import (
+	"sync"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/util/flowcontrol"
+
+	deprovisioningevents "github.com/aws/karpenter-core/pkg/controllers/deprovisioning/events"
+	"github.com/aws/karpenter-core/pkg/events"
+	"github.com/aws/karpenter-core/pkg/test"
+)
+
+var eventRecorder events.Recorder
+var internalRecorder *InternalRecorder
+
+type InternalRecorder struct {
+	mu    sync.RWMutex
+	calls map[string]int
+}
+
+func NewInternalRecorder() *InternalRecorder {
+	return &InternalRecorder{
+		calls: map[string]int{},
+	}
+}
+
+func (i *InternalRecorder) Event(_ runtime.Object, _, reason, _ string) {
+	i.mu.Lock()
+	defer i.mu.Unlock()
+	i.calls[reason]++
+}
+
+func (i *InternalRecorder) Eventf(object runtime.Object, eventtype, reason, messageFmt string, _ ...interface{}) {
+	i.Event(object, eventtype, reason, messageFmt)
+}
+
+func (i *InternalRecorder) AnnotatedEventf(object runtime.Object, _ map[string]string, eventtype, reason, messageFmt string, _ ...interface{}) {
+	i.Event(object, eventtype, reason, messageFmt)
+}
+
+func (i *InternalRecorder) Calls(reason string) int {
+	i.mu.RLock()
+	defer i.mu.RUnlock()
+	return i.calls[reason]
+}
+
+func TestRecorder(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "EventRecorder")
+}
+
+var _ = BeforeEach(func() {
+	internalRecorder = NewInternalRecorder()
+	eventRecorder = events.NewRecorder(internalRecorder)
+	events.PodNominationRateLimiter = flowcontrol.NewTokenBucketRateLimiter(5, 10)
+})
+
+var _ = Describe("Event Creation", func() {
```

Comment:
Is this needed? I'm assuming that the event suite_test should be enough to validate the event creation behavior

Reply:
Eh, I was thinking it'd be nice to have, but you're right that really we're only just creating an event and returning it. I'll remove.